src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts

```diff
@@ -89,6 +91,17 @@ export class ServicesComponent extends ListWithDetails implements OnChanges, OnI
   }
 
   ngOnInit() {
+    this.tableActions.unshift({
+      permission: 'create',
+      icon: Icons.add,
+      click: () =>
+        this.clusterCreation
+          ? (this.bsModalRef = this.modalService.show(ServiceFormComponent))
+          : this.router.navigate([BASE_URL, { outlets: { modal: [URLVerbs.CREATE] } }]),
+      name: this.actionLabels.CREATE,
+      canBePrimary: (selection: CdTableSelection) => !selection.hasSelection,
+      disable: (selection: CdTableSelection) => this.getDisable('create', selection)
+    });
```

Comment:
Same thing here: rather than internally modifying the behaviour of this component based on who's calling it (a kind of responsibility inversion principle), why not extending the `ListWithDetails` base component to support this tuning?

Reply:
I tried extending the ListWithDetails component to support this thing, the issue with that is - this class is being extended by almost every other component such as rbd-list, osd-list etc. and there were certain methods which we need to modify in our case, but in-fact they affect other components as well. So that would require a huge refactoring. Since we are following the same approach in host component as well. Maybe we can make this change in the Followups PR.

Reply:
@aaSharma14 I was working to change this in a different way rather than depending upon the `clusterCreation` flag. If it looks good you can also use that instead of the `ListWithDetails`. What do you think? @epuertat ?
https://github.com/ceph/ceph/pull/43041/commits/278d79ef58b1cf3bcac23bc5f5305cae7c50f4d2#diff-207940d141308083bee48d8b8e735b0a89caca64de81c5cc25e53447df705801R110-R119

Reply:
The reason I am doing it like this is because, this change is not specific to the cluster creation. Its more of a routed modal thing. We only want the routed modal in the original component. We don't want to routed modal in the cluster expansion wizard. We just need a simple normal modal.